src/d/parser/dmodule.d

```diff
@@ -17,10 +17,12 @@ auto parseModule(ref TokenRange trange) {
 
 	Name name;
 	Name[] packages;
+	Location locationName;
 
 	if (trange.front.type == TokenType.Module) {
 		trange.popFront();
 		name = trange.front.name;
+		locationName = trange.front.location;
```

Comment:
Shouldn't this point to the whole module declaration? And then, the name we are looking for is `moduleDeclarationLocation`

Reply:
Well, do we want it to underline…
a) the whole module declaration or
b) only the wrong package+name identifier combo?

Reply:
```
// a)
module wrong_name;
       ~~~~~~~~~~
       ^

// b)
module wrong_name;
~~~~~~~~~~~~~~~~~~
^
```